src/services/offer.js

```diff
@@ -192,29 +195,64 @@ class OfferService {
 
     /**
      * Fetches offers according to specified options
+     * Learn more about keyset search here: https://github.com/NIAEFEUP/nijobs-be/issues/129
+     *
      * @param {*} options
      * value: Text to use in full-text-search
-     * offset: Point to start looking (and limiting)
+     * lastOfferId: Id of the last fetched offer
      * limit: How many offers to show
      * jobType: Array of jobTypes allowed
      */
-    get({ value = "", offset = 0, limit = OfferService.MAX_OFFERS_PER_QUERY, showHidden = false, showAdminReason = false, ...filters }) {
-
-        const offers = (value ? Offer.find(
-            { "$and": [this._buildFilterQuery(filters), { "$text": { "$search": value } }] }, { score: { "$meta": "textScore" } }
-        ) : Offer.find(this._buildFilterQuery(filters))).current();
+    async get({ value = "", lastOfferId = null, limit = OfferService.MAX_OFFERS_PER_QUERY,
+        showHidden = false, showAdminReason = false, ...filters }) {
+
+        let offers;
+        if (lastOfferId && value) {
+            const lastOfferScore = await this.getTextSearchScoreById(lastOfferId, value);
+
+            offers = Offer.aggregate([
+                { $match: { $text: { $search: value } } },
+                { $addFields: {
+                    score: { $meta: "textScore" },
+                    adminReason: { $cond: [showAdminReason, "$adminReason", "$$REMOVE"] }
+                } },
+                { $match: { "$or": [
+                    { score: { "$lt": lastOfferScore } },
+                    { score: lastOfferScore, _id: { "$gt": ObjectId(lastOfferId) } }
```

Comment:
What if there are more offers now that have a bigger score than the last score? They wouldn't be returned

Reply:
Also, why do we care about the score? Can't we just sort by the id?

Reply:
If we do that, the offers best matching the search won't be at the top, which isn't ideal

Reply:
Also, do we want new offers to be returned in the infinite scroll? That's not the approach I was following. In this article, they prioritize consistency over new items: https://medium.com/swlh/mongodb-pagination-fast-consistent-ece2a97070f3

Reply:
Ok, leave it as is then